ferveo-python/ferveo/__init__.pyi

```diff
@@ -192,3 +192,71 @@ def decrypt_with_shared_secret(
         dkg_params: DkgPublicParameters,
 ) -> bytes:
     ...
+
+
+class ThresholdEncryptionError(Exception):
+    pass
+
+
+class InvalidShareNumberParameter(Exception):
+    pass
+
+
+class InvalidDkgStateToDeal(Exception):
+    pass
+
+
+class InvalidDkgStateToAggregate(Exception):
+    pass
+
+
+class InvalidDkgStateToVerify(Exception):
+    pass
+
+
+class InvalidDkgStateToIngest(Exception):
+    pass
```

Comment:
Aside from the use of Python base exceptions eg. `ValueError` (thanks for adding that 👍 ), my other thought there was around other logical groupings. For example, this group of exceptions can all potentially subclass a common parent.

So for these highlighted exceptions something like the following for example:
```suggestion
class InvalidDkgStateException(Exception):
    pass
        
class InvalidDkgStateToDeal(InvalidDkgStateException):
    pass


class InvalidDkgStateToAggregate(InvalidDkgStateException):
    pass


class InvalidDkgStateToVerify(InvalidDkgStateException):
    pass


class InvalidDkgStateToIngest(InvalidDkgStateException):
    pass
```

Perhaps something similar can be done for the Dealer exceptions below where the base class is

```
class DealerException(Exception):
    pass
```

and the other Dealer Exceptions subclass it.

Perhaps there are other similar groupings that can be done. Would these additional Iogical groupings also make sense for what we are trying to achieve? wdyt?